### PR TITLE
Remove iaf_neuron

### LINKED ISSUE #21: difference between iaf_neuron and iaf_psc_alpha unclear
Both iaf_neuron and iaf_psc_alpha are implementations of a leaky integrate-and-fire model with alpha-function shaped synaptic currents, but the documentation differs between these and does not explain why both models are needed.

https://trac.nest-initiative.org/trac/wiki/CodeReorganisationJune2006 states: "iaf_neuron has been retired. Use iaf_psc_alpha instead."

### PR DESCRIPTION
**Pull request by jougs**
This PR replaces all uses of `iaf_neuron` by `iaf_psc_alpha` and removes `iaf_neuron` completely from the NEST sources.

This is possible because the two were basically the same model. The only difference was the number of synaptic time constants. `iaf_neuron` only had a single _tau_syn_, while `iaf_psc_alpha` has _tau_syn_ex_ for excitatory synapses and _tau_syn_in_ for inhibitory synapses. The model was removed to avoid confusion and make simulation scripts more concise.

This PR resolves #21.

### CODE REVIEW

**Comment by heplesser** [comment]
@jougs From a quick check it appears that you have done a very thorough job, but could you confirm explicitly that all examples still work?

I also think that we might want to include this only in NEST 3, but not in NEST 2.12, since it is a non-trivial change to the NEST user experience.

**Comment by heplesser** [comment]
@abigailm Could you be reviewer number two on this one?

**Comment by abigailm** [comment]
👍 from me, if Jochen can confirm all the examples still work

File: extras/userdoc/md/documentation/quickref.md [hunk 1]
```diff
@@ -174,7 +174,7 @@ Create a nested multi-dimensional network structure in the current subnet.
 
 ### Neuron Models
 
-[iaf\_neuron](../helpindex/cmds/iaf_neuron),
+[iaf\_neuron](../helpindex/cmds/iaf_psc_alpha),
```
Comment (steffengraber) [inline extras/userdoc/md/documentation/quickref.md]: Could't we delete this line? 

    [iaf\_neuron](../helpindex/cmds/iaf_psc_alpha)

File: extras/userdoc/md/documentation/parallel-computing.md [hunk 2]
```diff
@@ -59,7 +59,7 @@ minimize their interaction. Devices thus do not have proxies on remote virtual
 processes.
 
 The node distribution for a small network consisting of `spike_generator`, four
-`iaf_neuron`s, and a `spike_detector` in a scenario with two processes with two
+`iaf_psc_alpha`s, and a `spike_detector` in a scenario with two processes with two
 threads each is shown in the following figure:
 
 ![Node\_distribution](../../img/Node_distribution.png)
```
Comment (steffengraber) [inline extras/userdoc/md/documentation/parallel-computing.md]: See line 66: `iaf=iaf\_neuron`

File: extras/userdoc/md/documentation/quickref.md (hunk 1 above)
Comment (jougs) [inline extras/userdoc/md/documentation/quickref.md]: Yes, definitely. These were not matched by the regular expression I used for checking. Done in 3b9fb12f.

File: extras/userdoc/md/documentation/parallel-computing.md (hunk 2 above)
Comment (jougs) [inline extras/userdoc/md/documentation/parallel-computing.md]: See above.

**Comment by jougs** [comment]
@stinebuu: I saw those as well and first thought they're still OK. Now I reconsidered and changed them in 6926fa0e4.

**Comment by jougs** [comment]
Now that 2.14 is officially released and announced, I don't see any reason to delay this any further.

**Comment by heplesser** [comment]
@jougs Agreed!

**Pull request merged**